src/Symfony/Component/Routing/Generator/CompiledUrlGenerator.php

```diff
@@ -53,6 +52,16 @@ public function generate($name, $parameters = [], $referenceType = self::ABSOLUT
 
         list($variables, $defaults, $requirements, $tokens, $hostTokens, $requiredSchemes) = $this->compiledRoutes[$name];
 
+        if (isset($defaults['_canonical_route']) && isset($defaults['_locale'])) {
```

Comment:
This if = we are in i18n routing. Checking isset($defaults['_locale']) is a safety.